src/gui/sdlmain.cpp

```diff
@@ -1095,6 +1095,25 @@ static void setup_presentation_mode(FRAME_MODE &previous_mode)
 		schedule_synced();
 }
 
+static void NewMouseScreenParams()
+{
+	int abs_x, abs_y;
+	SDL_GetMouseState(&abs_x, &abs_y);
+	if (abs_x < 0 || abs_x > UINT16_MAX || abs_y < 0 || abs_y > UINT16_MAX) {
+		LOG_ERR("SDL: Unexpected mouse position values: %d, %d", abs_x, abs_y);
+		abs_x = std::clamp(abs_x, 0, static_cast<int>(UINT16_MAX));
+		abs_y = std::clamp(abs_y, 0, static_cast<int>(UINT16_MAX));
+	}
```

Comment:
This is a hefty sledge hammer to deal with SDL's possibly wild integer mouse coordinates (and rightly so.. ).

The values start life without any guarantees of being absolute values, but we do expect it.

Would something like this be worth replacing:
``` c++
int x = 0;
int y = 0;
SDL_GetMouseState(&x, &y);

const auto abs_x = check_cast<uint16_t>(x);
const auto abs_y = check_cast<uint16_t>(y);
```

I wouldn't want actual cursor positions to be clamped anyway, so if SDL is giving us negative values or in the millions, I'd rather be notified in debug-mode versus users complaining about "phantom edges" and boundaries when moving the mouse.

Reply:
Turned checks into asserts, as agreed in the remark below.